### ISSUE TITLE
solid/reactivity rule does not report nested signal call in non tracked scope

### ISSUE THREAD

**Issue by Tronikelis** [solidjs-community/eslint-plugin-solid#124]
**Describe the bug**
Best way to explain is an example

https://playground.solidjs.com/anonymous/695d5961-9a9d-4422-8d72-e1d8f06cf98e

```tsx
function useExample(input: number | number[]) {}

function Counter(props: ComponentProps<"button">) {
  const [c, setC] = createSignal(1);
  
  // reports, nice
  useExample(c());
  // does not report
  useExample([c()]);

  return <div>hello</div>;
}
```

**Expected behavior**
Should report for both of them

**Environment (please complete the following information):**

   - OS: Windows 10
   - Node version 20.10.0
   - eslint-plugin-solid version 0.13.1
   - eslint version 8.56.0

- [ ] I would be willing to contribute a PR to fix this issue

**Comment by joshwilsonvu** [comment]
Hmm, thanks for this. Looks like in an attempt to allow code like

```tax
useExample([() => c()])
```

I inadvertently allowed code like `useExample([c()])`. Will just need to tighten up the checks for custom hooks—they're meant to be pretty permissive, but definitely not meant to allow incorrect code like this.

**Comment by Tronikelis** [comment]
thanks
